assets/js/googlesitekit/datastore/user/surveys.js

```diff
@@ -0,0 +1,181 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import isPlainObject from 'lodash/isPlainObject';
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { createFetchStore } from '../../data/create-fetch-store';
+import { createValidatedAction } from '../../data/utils';
+import { createCacheKey } from '../../api';
+import { getItem, setItem } from '../../api/cache';
+
+const fetchTriggerSurveyStore = createFetchStore( {
+	baseName: 'triggerSurvey',
+	controlCallback: ( { triggerID } ) => {
+		return API.set( 'core', 'user', 'survey-trigger', { triggerID } );
+	},
+	argsToParams: ( triggerID ) => {
+		return { triggerID };
+	},
+	reducerCallback: ( state, { survey_payload, session } ) => { // eslint-disable-line camelcase
+		// We don't replace survey if we already have one.
+		if ( state.currentSurveySession || state.currentSurvey ) {
+			return state;
+		}
+		return {
+			...state,
+			currentSurvey: survey_payload,
+			currentSurveySession: session,
+		};
+	},
+	validateParams: ( { triggerID } = {} ) => {
+		invariant( 'string' === typeof triggerID && triggerID.length, 'triggerID is required and must be a string' );
+	},
+
+} );
+
+const fetchSendSurveyEventStore = createFetchStore( {
+	baseName: 'sendSurveyEvent',
+	controlCallback: ( { event, session } ) => API.set( 'core', 'user', 'survey-event', { event, session } ),
+	argsToParams: ( event, session ) => {
+		return { event, session };
+	},
+} );
+
+const baseInitialState = {
+	currentSurvey: null,
+	currentSurveySession: null,
+};
+
+const baseActions = {
+	/**
+	 * Triggers a survey.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} triggerID     Trigger ID for the survey.
+	 * @param {Object} options       Survey options.
+	 * @param {number} [options.ttl] Optional. TTL for survey.
+	 * @return {Object} Object with `response` and `error`.
+	 */
+	triggerSurvey: createValidatedAction(
+		( triggerID, options = {} ) => {
+			const { ttl = 0 } = options;
+			invariant( 'string' === typeof triggerID && triggerID.length, 'triggerID is required and must be a string' );
+			invariant( isPlainObject( options ), 'options must be an object' );
+			invariant( 'number' === typeof ttl, 'options.ttl must be a number' );
+		},
+		function* ( triggerID, options = {} ) {
+			const { ttl = 0 } = options;
+			const { select } = yield Data.commonActions.getRegistry();
+			if ( null !== select( STORE_NAME ).getCurrentSurvey() ) {
+				return;
+			}
+			const cacheKey = createCacheKey( 'core', 'user', 'survey-event', { triggerID } );
+			const { cacheHit } = yield Data.commonActions.await( getItem( cacheKey ) );
+			if ( false === cacheHit && ttl ) {
```

Comment:
I'm finding this all a bit dense to read without empty lines. Any opposition to:

```suggestion
			const { select } = yield Data.commonActions.getRegistry();

			if ( null !== select( STORE_NAME ).getCurrentSurvey() ) {
				return;
			}

			const cacheKey = createCacheKey( 'core', 'user', 'survey-event', { triggerID } );
			const { cacheHit } = yield Data.commonActions.await( getItem( cacheKey ) );

			if ( false === cacheHit && ttl ) {
```